### PR TITLE
fix/ibc owner transfer coins

### PR DESCRIPTION
**Pull request by faisalnaveedRNS**
## Description

when a user buy NFT, token should be transfer to user, in this PR this issue of token transfer to owner was resolved for IBC tokens

Closes: #PS-63

complete_pending_execution was updated

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/pylons/types/coins.go [hunk 1]
```diff
@@ -35,12 +35,17 @@ func CookbookDenom(cookbookID, denom string) (string, error) {
 }
 
 // IsCookbookDenom checks if an inputted denom is a valid cookbookDenom
+// If denom is IBC denom we will not consider for CookBookDenom
 func IsCookbookDenom(denom string) bool {
 	split := strings.Split(denom, denomDivider)
 	if len(split) != 2 {
```
Comment (mustafapylons) [inline x/pylons/types/coins.go:41]: "2" can be in a constant variable

File: x/pylons/types/coins.go [hunk 2]
```diff
@@ -35,12 +35,17 @@ func CookbookDenom(cookbookID, denom string) (string, error) {
...
 		return false
 	}
 
+	// check if it is IBC denom
+	if IsIBCDenomRepresentation(denom) {
+		return false
+	}
 	// validate cookbook ID
 	err := ValidateID(split[0])
```
Comment (mustafapylons) [inline x/pylons/types/coins.go:50]: "0" can be in a constant variable

**Review by mustafapylons — COMMENTED** [review]
few comments, otherwise code is good.

**Review by BlackPanthar — COMMENTED** [review]
Reviewed, awaiting resolution to mustafa's comments before approval

**Review by mustafapylons — APPROVED** [review]
Looks good to me now.

**Review by BlackPanthar — APPROVED** [review]

**Review by mustafapylons — APPROVED** [review]
Looks good to me.

**Review by MikeSofaer — APPROVED** [review]

**Pull request merged**